### ISSUE TITLE
Rules MD044 and MD063 can be made to conflict

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#432]
### rumdl version

0.1.22

### Installation method

mise

### Bug description

While setting up a scenario to explain https://github.com/rvben/rumdl/issues/431, I found a bug. It's possible to configure rules MD044 and MD063 so that `rumdl` can't fix it.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   # Install Good Application on your computer

   Install Good Application on your computer.
   ```
1. Create this `rumdl` configuration:
   ```toml
   [MD044]
   names = ["Good Application"]

   [MD063]
   enabled = true
   style = "sentence-case"
   ```
1. Run `rumdl fmt` with that configuration on that Markdown.

### Expected behavior

Such configuration would be possible, possibly with MD063 taking into account MD044 settings as described in: https://github.com/rvben/rumdl/issues/431.

### Actual behavior

`rumdl` returns this error:

```shell
Warning: Failed to converge after 100 iterations.
This likely indicates a bug in rumdl.
Rule codes: MD063, MD044
Please report at: https://github.com/rvben/rumdl/issues/new?template=bug_report.yml
test.md:1:3: [MD063] Heading should use sentence case: 'Install Good Application on your computer' -> 'Install good application on your computer'
```

### Configuration

```toml
[MD044]
names = ["Good Application"]

[MD063]
enabled = true
style = "sentence-case"
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.23 as part of the same fix for #431. MD063 now respects MD044 proper names, so the two rules no longer conflict and the convergence loop is gone.
